mm2src/coins/solana.rs

```diff
@@ -233,10 +234,10 @@ pub async fn solana_coin_with_policy(
 /// pImpl idiom.
 pub struct SolanaCoinImpl {
     ticker: String,
-    key_pair: SolKeypair,
+    pub key_pair: SolKeypair,
     client: RpcClient,
     decimals: u8,
-    my_address: String,
+    pub my_address: String,
     spl_tokens_infos: Arc<Mutex<HashMap<String, SplTokenInfo>>>,
     /// This spawner is used to spawn coin's related futures that should be aborted on coin deactivation
     /// and on [`MmArc::stop`].
```

Comment:
Do you really need to make these fields (plus abortable) `pub`?
I could pass clippy warnings check making them private
![Screenshot 2024-08-08 at 14 04 00](https://github.com/user-attachments/assets/85a7886f-6fef-4d04-9a72-0f4d6e3f4cb3)

also `amount_to_ui_amount` functions is never used. I think it could be useful in the future, please add dead code annotation.

Reply:
@laruh, yes, I was able to access the data using the existing methods, so there's no need to make them public.